### ISSUE TITLE
react-native init doesn't create any files when run from a path with special characters

### ISSUE THREAD

**Issue by mkonicek** [facebook/react-native#3474]
e.g.:

    cd /tmp/\[apps\]/
    react-native init AwesomeProject

Only package.json and node_modules are created, no ios or android folders.

**Comment by jaygarcia** [comment]
Confirmed.  Joy.  It even says:
```
To run your app on iOS:
   Open /www/ios/KeyGenMusicPlayer/[apps]/Test/ios/Test.xcodeproj in Xcode
   Hit Run button
To run your app on Android:
   Have an Android emulator running, or a device connected
   cd /www/ios/KeyGenMusicPlayer/[apps]/Test
   react-native run-android

```

**Comment by mkonicek** [comment]
Yup :)

**Comment by mihirsoni** [comment]
@mkonicek  This is true , but when I do with the local-cli from cloned RN code, It does work as expected but not with the NPM package  source code.

**Comment by rumal** [comment]
I tried to recreate this but ended up with this error. I tried removing node modules as well. No luck. 
I have npm linked the `react-native-cli` and `react-native` to the source and when `react-native-cli` creates the folder and try to run the local cli from the `react-native` project in the node_modules, I entered and link the react-native package as well.

````
 /tmp/[a] > react-native init AwesomeProject
prompt: Directory AwesomeProject already exist. Continue?:  (no) y
This will walk you through creating a new React Native project in /private/tmp/[a]/AwesomeProject
Installing react-native package from npm...
readline.js:924
            throw err;
            ^

ReferenceError: Unknown plugin "syntax-async-functions" specified in "foreign" at 0
    at /Users/pasinduperera/projects/react-native/node_modules/babel-core/lib/transformation/file/options/option-manager.js:181:17
    at Array.map (native)
    at Function.normalisePlugins (/Users/pasinduperera/projects/react-native/node_modules/babel-core/lib/transformation/file/options/option-manager.js:163:20)
    at OptionManager.mergeOptions (/Users/pasinduperera/projects/react-native/node_modules/babel-core/lib/transformation/file/options/option-manager.js:259:36)
    at compile (/Users/pasinduperera/projects/react-native/node_modules/babel-register/lib/node.js:87:15)
    at loader (/Users/pasinduperera/projects/react-native/node_modules/babel-register/lib/node.js:128:14)
    at Object.require.extensions.(anonymous function) [as .js] (/Users/pasinduperera/projects/react-native/node_modules/babel-register/lib/node.js:138:7)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:311:12)
    at Module.require (module.js:366:17)
    at require (module.js:385:17)
````

Further more I tried the same with the latest released (`9a64f17`) version and it works just fine and this issue is also not present. Any changes on this. I saw the tests are also commented out at 6766d5ed546cfa03a78577e12ed169deb1f1584b

**Comment by mkonicek** [comment]
Hi there! This issue is being closed because it has been inactive for a while.

But don't worry, it will live on with ProductPains! Check out its new home: https://productpains.com/post/react-native/react-native-init-doesnt-create-any-files-when-run-from-a-path-with-special-characters

ProductPains helps the community prioritize the most important issues thanks to its voting feature.
It is easy to use - just login with GitHub.

Also, if this issue is a bug, please consider sending a PR with a fix.
We're a small team and rely on the community for bug fixes of issues that don't affect fb apps.
